sv.c

```diff
@@ -5795,6 +5795,30 @@ Perl_newSV(pTHX_ const STRLEN len)
     }
     return sv;
 }
+
+/*
+=for apidoc newSVpvz
+
+Creates a new SV sized with enough preallocated space to hold a string of
+C<len> bytes (plus a trailing NUL).
+
+The SV is initialized with a zero-length string with C<SvPOK> set.
+
+The reference count for the new SV is set to 1.
+
+=cut
+*/
+SV *
+Perl_newSVpvz(pTHX_ const STRLEN len)
+{
+    SV *sv = newSV_type(SVt_PV);
+    sv_grow_fresh(sv, len + 1);
+    SvPOK_only(sv);
+    SvPVX(sv)[0] = '\0';
```

Comment:
Small thought: I wonder if this should also `SvPVX(sv)[len] = '\0'` so as to ensure that if the caller only initialises the `[0 .. len-1]` bytes to whatever was wanted, there's definitely still a trailing NUL at the end.